packages/app/obojobo-document-engine/src/scripts/oboeditor/components/editor-app.js

```diff
@@ -123,6 +161,47 @@ class EditorApp extends React.Component {
 		this.reloadDraft(this.state.draftId, mode)
 	}
 
+	displayLockedState() {
+		this.setState({
+			requestStatus: 'invalid',
+			requestError: {
+				title: 'Module is Being Edited.',
+				message:
+					'Someone else is currently editing this module. Try reloading this tab in a few minutes (' +
+					this.props.settings.editLocks.autoExpireMinutes +
+					' or more).'
+			}
+		})
+	}
+
+	startRenewEditLockInterval(draftId) {
+		if (this._isCreatingEditLock === true) return Promise.resolve()
+		this._isCreatingEditLock = true
+
+		return this.createEditLock(draftId, this.state.model.get('contentId'))
+			.then(() => {
+				// only create the lock interval when we've got a successful lock
+				clearInterval(this.renewLockInterval) // clear any existing interval
+				this.renewLockInterval = setInterval(() => {
+					this.createEditLock(draftId, this.state.model.get('contentId')).catch(() =>
+						this.displayLockedState()
+					)
+				}, this.editLocks.autoExpireMs)
+
+				this._isCreatingEditLock = false
+			})
+			.catch(() => {
+				this.displayLockedState()
+				this._isCreatingEditLock = false
+			})
+	}
```

Comment:
I gave it a shot but had some issues w/ it passing tests, it was weirdly clearing when running the 2nd time.  I'm scared so I'm keeping it.  I also simplified a little - so I it might be a tad better